packages/metals-languageclient/src/fetchMetals.ts

```diff
@@ -135,3 +172,64 @@ export function calcServerDependency(serverVersion: string): string {
     return `org.scalameta:metals_${binaryVersion}:${serverVersion}`;
   }
 }
+
+export async function fetchCoursier(
+  coursierFetchPath: string,
+  handleOutput: (out: Buffer) => void
+) {
+  async function runChainedCommands(
+    command: string[],
+    initValue?: Promise<Output> | undefined
+  ) {
+    const out = command.reduce((acc, curr) => {
+      const res = () => {
+        const commandArr = curr.split(" ");
+        return run(commandArr[0], commandArr.slice(1), handleOutput, {
+          cwd: coursierFetchPath,
+        });
+      };
+      return acc ? acc.then(() => res()) : res();
+    }, initValue);
+    return (await out)?.code;
+  }
+
+  if (process.platform == "win32") {
+    // Windows
+    return runChainedCommands([
+      "curl -fLo cs-x86_64-pc-win32.zi https://github.com/coursier/launchers/raw/master/cs-x86_64-pc-win32.zip",
```

Comment:
We should specify a version of coursier instead of latests, this will be safer if coursier breaks at some version